lib/agent.js

```diff
@@ -174,12 +174,14 @@ Agent.prototype.start = function (opts) {
     return this
   } else if (this._conf.logLevel === 'trace') {
     var _ancestors = ancestors(module)
-    var basedir = path.dirname(process.argv[1])
     var stackObj = {}
     Error.captureStackTrace(stackObj)
 
+    // Attempt to load package.json from process.argv.
+    var pkg = null
```

Comment:
I see we've pulled the `pkg` variable up to the outer scope for readability, and possibly [some side eye at `var` hoisting](https://developer.mozilla.org/en-US/docs/Glossary/Hoisting).